Linker error building on OSX 10.9.5

Encountered the following error while running `premake5 gmake && make all`:
```
Building configurations...
Running action 'gmake'...
Generated Makefile...
Generated test.make...
Generated connect.make...
Generated info.make...
Generated yojimbo.make...
Generated client.make...
Generated server.make...
Generated secure_server.make...
Generated client_server.make...
Generated soak.make...
Generated profile.make...
Generated simple_messages.make...
Done (43ms).
==== Building yojimbo (debug_x64) ====
Creating bin
Creating obj/x64/Debug/yojimbo
yojimbo.cpp
yojimbo_address.cpp
yojimbo_channel.cpp
yojimbo_client_server.cpp
yojimbo_common.cpp
yojimbo_connection.cpp
yojimbo_encryption.cpp
yojimbo_matcher.cpp
yojimbo_network.cpp
yojimbo_packet.cpp
yojimbo_packet_processor.cpp
yojimbo_platform.cpp
yojimbo_simulator.cpp
yojimbo_sockets.cpp
yojimbo_transport.cpp
Linking yojimbo
==== Building test (debug_x64) ====
Creating obj/x64/Debug/test
test.cpp
Linking test
ld: warning: directory not found for option '-L/usr/lib64'
ld: unknown option: --start-group
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [bin/test] Error 1
make: *** [test] Error 2
```

From what I can tell, there isn't a `lib64` and clang targeting Darwin doesn't support `--start-group`/`--end-group` on OSX.

`premake5 --version` outputs: `premake5 (Premake Build Script Generator) 5.0.0-alpha9`
`make -v` outputs: `GNU Make 3.81`
`clang -v` outputs:
```
Apple LLVM version 5.1 (clang-503.0.40) (based on LLVM 3.4svn)
Target: x86_64-apple-darwin13.4.0
Thread model: posix
```

--- a/source/yojimbo_channel.cpp
+++ b/source/yojimbo_channel.cpp
@@ -116,6 +116,15 @@ namespace yojimbo
 
                 messages = (Message**) YOJIMBO_ALLOCATE( allocator, sizeof( Message* ) * numMessages );
 
+                if ( !messages )
+                {
+                    // Out of memory. Leave the arm empty (numMessages = 0) so ChannelPacketData::Free
+                    // stays safe, then fail the read; the channel treats this as a serialize failure.
+                    numMessages = 0;
+                    yojimbo_printf( YOJIMBO_LOG_LEVEL_ERROR, "error: failed to allocate messages (SerializeOrderedMessages)\n" );
+                    return false;
+                }
+
                 for ( int i = 0; i < numMessages; ++i )
                 {
                     messages[i] = NULL;
@@ -200,6 +209,15 @@ namespace yojimbo
                 Allocator & allocator = messageFactory.GetAllocator();
 
                 messages = (Message**) YOJIMBO_ALLOCATE( allocator, sizeof( Message* ) * numMessages );
+
+                if ( !messages )
+                {
+                    // Out of memory. Leave the arm empty (numMessages = 0) so ChannelPacketData::Free
+                    // stays safe, then fail the read; the channel treats this as a serialize failure.
+                    numMessages = 0;
+                    yojimbo_printf( YOJIMBO_LOG_LEVEL_ERROR, "error: failed to allocate messages (SerializeUnorderedMessages)\n" );
+                    return false;
+                }
 
                 for ( int i = 0; i < numMessages; ++i )
                     messages[i] = NULL;

--- a/source/yojimbo_network_simulator.cpp
+++ b/source/yojimbo_network_simulator.cpp
@@ -126,8 +126,14 @@ namespace yojimbo
 
         int numPackets = 0;
 
-        for ( int i = 0; i < yojimbo_min( m_numPacketEntries, maxPackets ); ++i )
+        // Scan the whole ring, but never return more than maxPackets. Packets are stored at
+        // m_currentIndex across the entire buffer, so bounding the scan by maxPackets (rather
+        // than the output count) would strand any packet held in a slot past that index.
+        for ( int i = 0; i < m_numPacketEntries; ++i )
         {
+            if ( numPackets >= maxPackets )
+                break;
+
             if ( !m_packetEntries[i].packetData )
                 continue;
 

--- a/test.cpp
+++ b/test.cpp
@@ -449,6 +449,59 @@ void test_address()
     }
 }
 
+void test_network_simulator_drains_all_slots()
+{
+    // Regression: ReceivePackets used to scan only the first min(numEntries, maxPackets) ring
+    // slots. Packets are stored across the whole ring (at m_currentIndex), so when a caller
+    // passes maxPackets < numEntries, any packet held in a tail slot was never drained. It must
+    // scan the entire ring and only cap how many packets it returns per call.
+
+    const int NumEntries = 4;
+
+    NetworkSimulator sim( GetDefaultAllocator(), NumEntries, 0.0 );
+
+    // Negative loss/duplicate rates make those RNG checks impossible to fire (random_float is
+    // always >= 0), so sends are fully deterministic, and they mark the simulator active.
+    sim.SetPacketLoss( -1.0f );
+    sim.SetDuplicates( -1.0f );
+
+    uint8_t payload[8];
+    for ( int i = 0; i < NumEntries; ++i )
+    {
+        memset( payload, (uint8_t) i, sizeof( payload ) );
+        sim.SendPacket( i, payload, sizeof( payload ) );    // fills ring slots 0..NumEntries-1, "to" == slot
+    }
+
+    sim.AdvanceTime( 1.0 );     // past every delivery time (latency 0)
+
+    // Drain with maxPackets (2) smaller than the ring (4). Everything must come out across calls,
+    // including packets in the tail slots that the old code stranded.
+    bool seen[NumEntries] = {};
+    int totalReceived = 0;
+
+    for ( int iter = 0; iter < NumEntries + 2; ++iter )
+    {
+        uint8_t * packetData[2];
+        int packetBytes[2];
+        int to[2];
+        const int n = sim.ReceivePackets( 2, packetData, packetBytes, to );
+        check( n <= 2 );        // never returns more than maxPackets
+        for ( int i = 0; i < n; ++i )
+        {
+            check( to[i] >= 0 && to[i] < NumEntries );
+            check( !seen[to[i]] );      // no packet delivered twice
+            check( packetBytes[i] == (int) sizeof( payload ) );
+            seen[to[i]] = true;
+            totalReceived++;
+            YOJIMBO_FREE( sim.GetAllocator(), packetData[i] );
+        }
+    }
+
+    check( totalReceived == NumEntries );
+    for ( int i = 0; i < NumEntries; ++i )
+        check( seen[i] );       // tail slots (index >= maxPackets) were drained too
+}
+
 void test_bit_array()
 {
     const int Size = 300;
@@ -2932,6 +2985,7 @@ int main( int argc, char ** argv )
         RUN_TEST( test_crypto_aead_vectors );
         RUN_TEST( test_queue );
         RUN_TEST( test_address );
+        RUN_TEST( test_network_simulator_drains_all_slots );
         RUN_TEST( test_bit_array );
         RUN_TEST( test_sequence_buffer );
         RUN_TEST( test_allocator_tlsf );

